File: ios/Classes/AttributionInfoProvider.swift
import Metrix
import Flutter

class AttributionInfoStreamHandler: NSObject, FlutterStreamHandler {
    private var attributionInfoEventSink: FlutterEventSink? = nil

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        attributionInfoEventSink = events
        MetrixClient.setOnAttributionChangedListener { (data: AttributionData) in
            var attributionDataMap: [String: String] = [:]

            if (data.acquisitionAd != nil) {
                attributionDataMap["acquisitionAd"] = data.acquisitionAd
            }
            if (data.acquisitionAdSet != nil) {
                attributionDataMap["acquisitionAdSet"] = data.acquisitionAdSet
            }
            if (data.acquisitionCampaign != nil) {
                attributionDataMap["acquisitionCampaign"] = data.acquisitionCampaign
            }
            if (data.acquisitionSource != nil) {
                attributionDataMap["acquisitionSource"] = data.acquisitionSource
            }
            if (data.acquisitionSubId != nil) {
                attributionDataMap["acquisitionSubId"] = data.acquisitionSubId
            }
            attributionDataMap["attributionStatus"] = data.attributionStatus.rawValue

            self.attributionInfoEventSink?(attributionDataMap)
        }

        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        attributionInfoEventSink = nil
        return nil
    }
}

File: android/src/main/kotlin/ir/metrix/flutter/AttributionInfoProvider.kt
package ir.metrix.flutter

import io.flutter.plugin.common.EventChannel
import ir.metrix.attribution.OnAttributionChangeListener
import ir.metrix.attribution.AttributionData

class AttributionInfoStreamHandler : EventChannel.StreamHandler {
    private var attributionInfoEventSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        attributionInfoEventSink = events
        ir.metrix.attribution.MetrixAttribution.setOnAttributionChangedListener(
            object : OnAttributionChangeListener {
                override fun onAttributionChanged(attributionData: AttributionData) {
                    val attributionDataMap: Map<String, String> = mapOf(
                        "acquisitionAd" to (attributionData.acquisitionAd ?: ""),
                        "acquisitionAdSet" to (attributionData.acquisitionAdSet ?: ""),
                        "acquisitionCampaign" to (attributionData.acquisitionCampaign ?: ""),
                        "acquisitionSource" to (attributionData.acquisitionSource ?: ""),
                        "acquisitionSubId" to (attributionData.acquisitionSubId ?: ""),
                        "attributionStatus" to (attributionData.attributionStatus?.name ?: "")
                    )
                    
                    attributionInfoEventSink?.success(attributionDataMap)
                }
            })
    }

    override fun onCancel(arguments: Any?) {
        attributionInfoEventSink = null
    }
}